### PR TITLE
Add kafka-clients 3.x support

### PR DESCRIPTION
**Pull request by lujiajing1126**
### Fix signature change of the intercepted method in kafka-clients 3.2.x
- [ ] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.

Due to the upstream PR https://github.com/apache/kafka/pull/11046, kafka-clients 3.2.x has changed the signature of `pollForFetches`. The return type is `org.apache.kafka.clients.consumer.internals.Fetch` now.

This will cause an exception in SW agent 8.13.x

image

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Clickhouse CI seems failing due to env failing.
I ask @kezhenxu94 to check whether we could use latest ck client lib on maven central.

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java [hunk 1]
```diff
@@ -100,6 +100,11 @@ public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allA
         return ret;
     }
 
+    @SuppressWarnings({"unchecked"})
+    protected Map<TopicPartition, List<ConsumerRecord<?, ?>>> fetchRecords(Object retObj) {
+        return (Map<TopicPartition, List<ConsumerRecord<?, ?>>>) retObj;
+    }
+
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/KafkaConsumerInterceptor.java:107]: What is the point of this method?

File: test/plugin/scenarios/kafka-scenario/support-version.list [hunk 2]
```diff
@@ -25,3 +25,6 @@
 2.6.1
 2.7.0
 2.8.0
+3.0.2
+3.1.2
+3.2.3
```
Comment (wu-sheng) [inline test/plugin/scenarios/kafka-scenario/support-version.list:30]: It looks like you are adding supports to 3.x, not just 3.2.x. Please update the PR title.

Comment (lujiajing1126) [reply test/plugin/scenarios/kafka-scenario/support-version.list:30]: Fixed

**Comment by lujiajing1126** [comment]
> Clickhouse CI seems failing due to env failing. I ask @kezhenxu94 to check whether we could use latest ck client lib on maven central.

It seems the artifact issue only exists in 0.3.0, but not 0.3.1

**Comment by wu-sheng** [comment]
> > Clickhouse CI seems failing due to env failing. I ask @kezhenxu94 to check whether we could use latest ck client lib on maven central.
> 
> It seems the artifact issue only exists in 0.3.0, but not 0.3.1

The issue is, clickhouse lib has dependencies hosts on https://jitpack.io/ somehow, which is down for 2 days.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**